isign/provisioner.py

```diff
@@ -0,0 +1,115 @@
+        for provisioning_profile in provisioning_profiles:
+            entitlements = cls.extract_entitlements(provisioning_profile)
+            if 'application-identifier' not in entitlements:
+                raise Exception("Could not find application-identifier in entitlements from provisioning profile {}"
+                                .format(provisioning_profile))
+            app_id = entitlements['application-identifier']
+            if app_id in app_id_to_pprof:
+                raise Exception("At least 2 provisioning profiles target the same application identifier: "
+                                "{}, {}".format(provisioning_profile, app_id_to_pprof[app_id]))
+            app_id_to_pprof[app_id] = provisioning_profile
+            app_id_to_entitlements_info[app_id] = {
+                "path": provisioning_profile,
+                "entitlements": biplist.writePlistToString(entitlements)
+            }
+        return app_id_to_pprof, app_id_to_entitlements_info
+
+    @classmethod
+    def parse_entitlements(cls, entitlements_paths):
+        """
+           Parse additional entitlements files into a map of
+             application id -> entitlements as a string
+        """
+        app_id_to_entitlements_info = {}
+        for entitlements_path in entitlements_paths:
+            entitlements = biplist.readPlist(entitlements_path)
+            app_id = entitlements['application-identifier']
+            if app_id in app_id_to_entitlements_info:
+                raise Exception("At least 2 entitlements files target the same application identifier: "
+                                "{}, {}".format(entitlements_path, app_id_to_entitlements_info[app_id]['source']))
+            app_id_to_entitlements_info[app_id] = {
+                "path": entitlements_path,
+                "entitlements": biplist.readPlist(entitlements_path)
+            }
+        return app_id_to_entitlements_info
+
+    def __init__(self, provisioning_profiles, entitlements_paths):
+        log.debug("provisioner")
+        log.debug(provisioning_profiles)
+        log.debug(entitlements_paths)
+        (app_id_to_pprof, app_id_to_entitlements_info) = self.parse_provisioning_profiles(provisioning_profiles)
+        self.app_id_to_pprof = app_id_to_pprof
+        log.debug(app_id_to_pprof)
+        self.app_id_to_entitlements_info = app_id_to_entitlements_info
+        log.debug(self.app_id_to_entitlements_info)
+
+        # each provisioning profile should already have entitlements. But maybe you want to override them.
+        self.app_id_to_entitlements_info.update(self.parse_entitlements(entitlements_paths))
+        # Note, it appears that overriden entitlements paths can only be "less" than the ones in the provisioning
+        # profile. For instance, a property that in the provisioning profile is "foo.*" can be "foo.bar.baz" in the
+        # entitlements, but not the reverse. We could do a sanity check here for that!
```

Comment:
Could generalize that `IdentifierMatcher` loop I commented on above to check this.

Reply:
Yup, that's the plan, should is be necessary. Just add a `is_match` which tests for score > 0